### PR TITLE
Updated snellen eye exam with peripheral vision

### PR DESCRIPTION
**Pull request by Wakie87**
Rebased and updated from PR #66

### CODE REVIEW

File: contrib/forms/snellen_peripheral/table.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  `left_1`              text NOT NULL DEFAULT '',
+  `left_2`              text NOT NULL DEFAULT '',
+  `right_1`             text NOT NULL DEFAULT '',
+  `right_2`             text NOT NULL DEFAULT '',
+  `both_1`              text NOT NULL DEFAULT '',
+  `both_2`              text NOT NULL DEFAULT '',
+  `peripheral_r1`       text NOT NULL DEFAULT '',
+  `peripheral_r2`       text NOT NULL DEFAULT '',
+  `peripheral_l1`       text NOT NULL DEFAULT '',
+  `peripheral_l2`       text NOT NULL DEFAULT '',
```
Comment (bradymiller) [inline contrib/forms/snellen_peripheral/table.sql]: All these text fields need to be reduced to just TEXT (A innodb and modern database issue):
``````
`left_1`              text,

**Comment by Wakie87** [comment]
Also what are your thoughts about just making this an update to the original snellen form? not a completely new form. I know it's only in the contrib folder...?

File: contrib/forms/snellen_peripheral/table.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  `colors`              varchar(3) NOT NULL default 'UNK',
+  `monocular`           varchar(3) NOT NULL default 'UNK',  
+  `notes`               text NOT NULL DEFAULT '',
```
Comment (bradymiller) [inline contrib/forms/snellen_peripheral/table.sql]: text fields needs to be reduced to just TEXT

File: contrib/forms/snellen_peripheral/templates/general_new.html [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	</tr>	
+	<tr>
+		<td><b>{xl t="Peripheral(L)"}</b></td>
+		<td><input type='text' size='2px' maxlength="2" name='Peripheral_l1' value='{$data->get_Peripheral_l1()}'>degrees</td>
+		<td><input type='text' size='2px' maxlength="2" name='Peripheral_l2' value='{$data->get_Peripheral_l2()}'>degrees</td>
+	</tr>	
+	<tr>
+		<td><b>{xl t="Peripheral(R)"}</b></td>
+		<td><input type='text' size='2px' maxlength="2" name='Peripheral_r1' value='{$data->get_Peripheral_r1()}'>degrees</td>
+		<td><input type='text' size='2px' maxlength="2" name='Peripheral_r2' value='{$data->get_Peripheral_r2()}'>degrees</td>
```
Comment (bradymiller) [inline contrib/forms/snellen_peripheral/templates/general_new.html]: degrees translated in 4 lines above

File: contrib/forms/snellen_peripheral/templates/general_new.html [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	</tr>
+	<tr class='text'>
+		<td><textarea cols='80' rows='5' name='notes'>{$data->get_notes()}</textarea></td>
+	</tr>
+</table>
+
+<table>
+	<tr>
+		<td>
+			<input type="submit" name="Submit" value="Save Form">
```
Comment (bradymiller) [inline contrib/forms/snellen_peripheral/templates/general_new.html]: translate Save Form

File: contrib/forms/snellen_peripheral/templates/general_new.html [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		</td>
+		<td>
+			<a href="{$DONT_SAVE_LINK}" class="link">[Don't Save]</a>
```
Comment (bradymiller) [inline contrib/forms/snellen_peripheral/templates/general_new.html]: translate Don't Save

File: contrib/forms/snellen_peripheral/report.php [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+		</tr>
+		<tr>
+			<td><?php echo xlt('Peripheral(R)') ?></td>
+			<td><?php echo $data['peripheral_r1'] ? $data['peripheral_r1'] : "__"; ?>&deg;</td>
+			<td><?php echo $data['peripheral_r2'] ? $data['peripheral_r2'] : "__"; ?>&deg;</td>
+		</tr>		
+	</table>
+	<table border='0' cellpadding='0' cellspacing='0' class='text'>
+		<tr>
+			<br><td><b><?php echo xlt('Able to distinguish between: Red, Green, Amber colors') ?>:</b> <?php echo $data['colors']?></td>
```
Comment (bradymiller) [inline contrib/forms/snellen_peripheral/report.php]:
``````
echo text($data['colors'])

File: contrib/forms/snellen_peripheral/report.php [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+		</tr>
+	</table>
+	<table border='0' cellpadding='0' cellspacing='0' class='text'>
+		<tr>
+			<td><b><?php echo xlt('Monocular Vision') ?>:</b> <?php echo $data['monocular']?></td>
```
Comment (bradymiller) [inline contrib/forms/snellen_peripheral/report.php]:
``````
echo text($data['monocular'])

File: contrib/forms/snellen_peripheral/report.php [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+	</table>	
+	<?php if ($data['notes'] != '') {?>
+	</p>
+	
+	<table border='0' cellpadding='0' cellspacing='0' class='text'>
+		<tr class='text'>
+			<td><b><?php echo xlt('NOTES') ?></b></td>
+		</tr>
+		<tr class='text'>
+			<td><p align='left'><?php echo $data['notes']?>&nbsp;</p></td>
```
Comment (bradymiller) [inline contrib/forms/snellen_peripheral/report.php]:
``````
echo text($data['notes'])

**Comment by bradymiller** [comment]
Since it's different form original snellen form, especially lots of database changes, would keep it separate from the original.

**Comment by bradymiller** [comment]
Review done. Didn't test. If address issues and it tests ok, then would bring this in.

**Comment by Wakie87** [comment]
Added to codebase but will remove and reference in the Wiki as it does not work.

**Pull request closed without merging**